PrestaShop/PrestaShop
# [WIP] Migrate Configure > Advanced Parameters > Webservices

Linked to https://github.com/PrestaShop/prestafony-project/issues/37

⚠️  This is a WIP page as I need help from @mickaelandrieu for the webservice keys listing features.

<hr>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop |
| Description?  | Migrate BO page 'webservice' to Symfony using modern controllers and form handlers |
| Type?         | new feature |
| Category?     | BO |
| BC breaks?    | yes |
| Deprecations? | no |
| Fixed ticket? |  |
| How to test?  | See below |

### How to test
Login on backoffice, then navigate to "Webservice page". For example use left menu "Configure > Advanced Parameters > Webservice" link.

Check that:
- the webservice keys listing is working (sorting, pagination, filtering, search, bulk actions, link)
- the "Configuration" form to enable Webservice usage and CGI mode is working
- the "add a new / edit webservice key" page is working (you can access it using "add a new key" button or a "edit" button on one webservice key)

#### BC break
As the Controller and the Template are being moved, this will disable any override of the previous controller and template.

### Possible issues because of the migration

> URLs have changed

Previous URL:
- `/admin/index.php?controller=AdminWebservice`
New URLs:
- `/admin/configure/advanced/webservice/add`
- `/admin/configure/advanced/webservice/`


> Design have changed

The new page is powered by Symfony and uses the new admin theme. I tried to make it look as much as possible to the previous one but the design is slightly different (example: previous theme used bootstrap icons, new theme uses material icons).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9272)
